armi/reactor/blueprints/componentBlueprint.py

```diff
@@ -186,7 +186,7 @@ def shape(self, shape):
     mergeWith = yamlize.Attribute(type=str, default=None)
     area = yamlize.Attribute(type=float, default=None)
 
-    def construct(self, blueprint, matMods):
+    def construct(self, blueprint, matMods, inputHeightsConsideredHot):
```

Comment:
Okay, so in the `Block` class above this makes more sense to me. But the variable name here is strange. Because "input heights" makes sense if this Component is part of an `Assembly` in some way, sure.

But `Components` are used by the Mech team and we want to support non-pin-type reactors. (I understand we don't do great at this yet, but I'm working on it.)

What do you think?

Reply:
Also, maybe this new parameter should have a default value?  What do you think?

Reply:
This parameter and its name are directly reflective of the case setting of the same name. I tried to reflect that intention in the docstring, but maybe I could be more clear there.

When Jake put up the draft PR, he was originally passing an entire case settings object down to this level. I did away with that because I find it frustrating when I have to whip up an entire `cs` just for a single boolean. 

So I made the name for this parameter exactly the same as the case setting whose value should be passed down here. I thought that made things simpler, but if you think it'd be more clear either (1) with a different name, or (2) if we passed the entire `cs`, then I am open to those options.

Reply:
With respect to a default value...

I'm not sure how I feel there. 

On the one hand, we could potentially make the default be the same as the default for the `inputHeightsConsideredHot` case setting (`True`). On the other hand, I really don't like that because the default on that case setting is the opposite of our typical workflow at this point in time (i.e. having axial expansion turned on).

I think my ideal solution would be if there were a way to not have to pass the value via method calls. But the only way I could conceive to do that is to store a `cs` on the `ComponentBlueprint`, and that is something I really don't want to do.

Maybe you see a clean way to get around all this?

Reply:
It is definitely better to pass one value instead of the entire settings object. For sure. Nice.

I suppose "input dimensions considered hot" would have been a more accurate name for that setting. But the people who created it were definitely thinking of pin-type reactors.